MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -8,60 +8,102 @@
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "The Avengers",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "Deadpool",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "The Green Knight",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: true),
+//        QuizQuestion(
+//            image: "Old",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "The Ice Age Adventures of Buck Wild",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "Tesla",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false),
+//        QuizQuestion(
+//            image: "Vivarium",
+//            text: "Рейтинг этого фильма больше чем 6?",
+//            correctAnswer: false)
+//    ]
+    
+    private let moviesLoader: MoviesLoading
+    private var movies: [MostPopularMovie] = []
     weak var delegate: QuestionFactoryDelegate?
     
-    init(delegate: QuestionFactoryDelegate?) {
+    init(moviesLoader: MoviesLoading, delegate: QuestionFactoryDelegate?) {
+        self.moviesLoader = moviesLoader
         self.delegate = delegate
     }
     
     func requestNextQuestion() {
-        guard let index = (0..<questions.count).randomElement() else {
-            delegate?.didRecieveNextQuestion(question: nil)
-            return
+        DispatchQueue.global().async { [weak self] in
+            guard let self = self else { return }
+            let index = (0..<self.movies.count).randomElement() ?? 0
+            
+            guard let movie = self.movies[safe: index] else { return }
+            
+            var imageData = Data()
+            
+            do {
+                imageData = try Data(contentsOf: movie.resizedImageURL)
+            } catch {
+                print("Failed to load image")
+            }
+            
+            let rating = Float(movie.rating) ?? 0
+            let randomRatingBound = Float((3...9).randomElement() ?? 0)
+            let text = "Рейтинг этого фильма больше чем \(String(format: "%.0f", randomRatingBound))?"
+            let correctAnswer = rating > randomRatingBound
+            
+            let question = QuizQuestion(
+                image: imageData,
+                text: text,
+                correctAnswer: correctAnswer)
+            
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+                self.delegate?.didRecieveNextQuestion(question: question)
+            }
+        }
+    }
+    
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Контекст захватывается сильно. В данном приложении это не вызовет утечек, однако это потенциально будущая ошибка.
```suggestion
        moviesLoader.loadMovies { [weak self] result in
```